metriq_gym/benchmarks/benchmark.py

```diff
@@ -1,17 +1,31 @@
 import argparse
-from typing import Any
 
-from metriq_gym.job_manager import JobManager
+from pydantic import BaseModel
+from dataclasses import dataclass
+
+from qbraid import QuantumDevice, ResultData
+
+
+@dataclass
+class BenchmarkData:
+    provider_job_id: list[str]
 
 
 class Benchmark:
-    def __init__(self, args: argparse.Namespace, params: dict[str, Any], job_manager: JobManager):
+    def __init__(
+        self,
+        args: argparse.Namespace,
+        params: BaseModel,
+    ):
         self.args = args
-        self.params = params
-        self.job_manager = job_manager
+        self.params: BaseModel = params
 
-    def dispatch_handler(self) -> None:
+    def dispatch_handler(self, device: QuantumDevice) -> BenchmarkData:
         raise NotImplementedError
 
-    def poll_handler(self) -> None:
+    def poll_handler(
```

Comment:
[nit] Does this not all fit < 100 chars?
```suggestion
        def poll_handler(self, job_data: BenchmarkData, result_data: list[ResultData]) -> None:
```